[Test Case Failure]: tests/llms/AI21StudioModel_test.py::test_no_system_context FAILED

### Test Case

 tests/llms/AI21StudioModel_test.py::test_no_system_context AND OTHER TESTS

### Root Cause

from logs:
```
tests/llms/AI21StudioModel_test.py::test_no_system_context 
-------------------------------- live log call ---------------------------------
HTTP Request: POST https://api.ai21.com/studio/v1/chat/completions "HTTP/1.1 422 Unprocessable Entity"
Calling POST https://api.ai21.com/studio/v1 failed with a non-200 response code: 422
FAILED                                                                   [ 32%]
tests/llms/AI21StudioModel_test.py::test_nonpreamble_system_context 
-------------------------------- live log call ---------------------------------
HTTP Request: POST https://api.ai21.com/studio/v1/chat/completions "HTTP/1.1 422 Unprocessable Entity"
Calling POST https://api.ai21.com/studio/v1 failed with a non-200 response code: 422
FAILED                                                                   [ 32%]
tests/llms/AI21StudioModel_test.py::test_preamble_system_context 
-------------------------------- live log call ---------------------------------
HTTP Request: POST https://api.ai21.com/studio/v1/chat/completions "HTTP/1.1 422 Unprocessable Entity"
Calling POST https://api.ai21.com/studio/v1 failed with a non-200 response code: 422
FAILED                                                                   [ 32%]
tests/llms/AI21StudioModel_test.py::test_multiple_system_contexts 
-------------------------------- live log call ---------------------------------
HTTP Request: POST https://api.ai21.com/studio/v1/chat/completions "HTTP/1.1 422 Unprocessable Entity"
Calling POST https://api.ai21.com/studio/v1 failed with a non-200 response code: 422
FAILED 

```



### Potential Solutions

modify the AI21StudioModel to address the 422 error

### Additional Context (optional)

_No response_

### Type of Test Case

Unit

## logs
```
FAILED tests/llms/AI21StudioModel_test.py::test_no_system_context - ai21.errors.UnprocessableEntity: Failed with http status code: 422 (UnprocessableEntity). Details: {"detail":"value is not a valid enumeration member; permitted: 'jamba-instruct', 'jamba-instruct-preview', 'jamba-next', 'jamba-large-next', 'jamba-large-next-2', 'jamba-1.5', 'jamba-1.5-large'"}

FAILED tests/llms/AI21StudioModel_test.py::test_nonpreamble_system_context - ai21.errors.UnprocessableEntity: Failed with http status code: 422 (UnprocessableEntity). Details: {"detail":"value is not a valid enumeration member; permitted: 'jamba-instruct', 'jamba-instruct-preview', 'jamba-next', 'jamba-large-next', 'jamba-large-next-2', 'jamba-1.5', 'jamba-1.5-large'"}

FAILED tests/llms/AI21StudioModel_test.py::test_preamble_system_context - ai21.errors.UnprocessableEntity: Failed with http status code: 422 (UnprocessableEntity). Details: {"detail":"value is not a valid enumeration member; permitted: 'jamba-instruct', 'jamba-instruct-preview', 'jamba-next', 'jamba-large-next', 'jamba-large-next-2', 'jamba-1.5', 'jamba-1.5-large'"}

FAILED tests/llms/AI21StudioModel_test.py::test_multiple_system_contexts - ai21.errors.UnprocessableEntity: Failed with http status code: 422 (UnprocessableEntity). Details: {"detail":"value is not a valid enumeration member; permitted: 'jamba-instruct', 'jamba-instruct-preview', 'jamba-next', 'jamba-large-next', 'jamba-large-next-2', 'jamba-1.5', 'jamba-1.5-large'"}
```

## solution
updated allowed_models to use this list: 

```python
['jamba-instruct', 'jamba-instruct-preview', 'jamba-next', 'jamba-large-next', 'jamba-large-next-2', 'jamba-1.5', 'jamba-1.5-large']
```

set the default model to `jamba-instruct`

After changing the default model to jamba-instruct,  the response status changed to `200`.
However a certain attribute errors was being caused

```
=========================================================================== short test summary info ============================================================================
FAILED tests/llms/AI21StudioModel_test.py::test_no_system_context - AttributeError: 'ChatCompletionResponse' object has no attribute 'content'
FAILED tests/llms/AI21StudioModel_test.py::test_nonpreamble_system_context - AttributeError: 'ChatCompletionResponse' object has no attribute 'content'
FAILED tests/llms/AI21StudioModel_test.py::test_preamble_system_context - AttributeError: 'ChatCompletionResponse' object has no attribute 'content'
FAILED tests/llms/AI21StudioModel_test.py::test_multiple_system_contexts - AttributeError: 'ChatCompletionResponse' object has no attribute 'content'
```
I changed the destructuring of the response and this fixed the issue.

--- a/swarmauri/standard/llms/concrete/AI21StudioModel.py
+++ b/swarmauri/standard/llms/concrete/AI21StudioModel.py
@@ -1,92 +1,91 @@
 import json
 from typing import List, Dict, Literal
-import ai21 
-from ai21.models.chat import ChatMessage 
+import ai21
+from ai21.models.chat import ChatMessage
 from swarmauri.core.typing import SubclassUnion
 
 from swarmauri.standard.messages.base.MessageBase import MessageBase
 from swarmauri.standard.messages.concrete.AgentMessage import AgentMessage
 from swarmauri.standard.llms.base.LLMBase import LLMBase
 
+
 class AI21StudioModel(LLMBase):
     api_key: str
-    allowed_models: List[str] = ['j2-light', 
-                                 'j2-light-instruct', 
-                                 'j2-mid', 
-                                 'j2-mid-instruct', 
-                                 'j2-ultra', 
-                                 'j2-ultra-instruct', 
-                                 'j2-grande-chat', 
-                                 'j2-jumbo-chat',
-                                 'jamba-instruct'
-                                 ]
-    name: str = "j2-light"
-    type: Literal['AI21StudioModel'] = 'AI21StudioModel'
+    allowed_models: List[str] = [
+        "jamba-instruct",
+        "jamba-instruct-preview",
+        "jamba-next",
+        "jamba-large-next",
+        "jamba-large-next-2",
+        "jamba-1.5",
+        "jamba-1.5-large",
+    ]
+    name: str = "jamba-instruct"
+    type: Literal["AI21StudioModel"] = "AI21StudioModel"
 
-    def _format_messages(self, messages: List[SubclassUnion[MessageBase]]) -> List[Dict[str, str]]:
-       # Get only the properties that we require
+    def _format_messages(
+        self, messages: List[SubclassUnion[MessageBase]]
+    ) -> List[Dict[str, str]]:
+        # Get only the properties that we require
         message_properties = ["content", "role"]
 
         # Exclude FunctionMessages
-        formatted_messages = [ChatMessage(content=message.content, role=message.role) for message in messages if message.role != 'system']
+        formatted_messages = [
+            ChatMessage(content=message.content, role=message.role)
+            for message in messages
+            if message.role != "system"
+        ]
         return formatted_messages
 
     def _get_system_context(self, messages: List[SubclassUnion[MessageBase]]) -> str:
         system_context = None
         for message in messages:
-            if message.role == 'system':
+            if message.role == "system":
                 system_context = message.content
         return system_context
 
-    
-    def predict(self, 
-        conversation, 
-        temperature=0.7, 
-        max_tokens=256, 
-        top_p=1.0, 
-        stop='\n', 
-        n=1, 
-        stream=False): 
-        ''' 
-        Args: 
+    def predict(
+        self,
+        conversation,
+        temperature=0.7,
+        max_tokens=256,
+        top_p=1.0,
+        stop="\n",
+        n=1,
+        stream=False,
+    ):
+        """
+        Args:
             top_p (0.01 to 1.0)  Limit the pool of next tokens in each step to the top {top_p*100} percentile of possible tokens
-            stop: stop the message when model generates one of these strings. 
+            stop: stop the message when model generates one of these strings.
             n: how many chat responses to generate
-            stream: whether or not to stream the result, for long answers. if set to true n must be 1. 
-        '''
+            stream: whether or not to stream the result, for long answers. if set to true n must be 1.
+        """
 
         # Create client
         client = ai21.AI21Client(api_key=self.api_key)
-        
+
         # Get system_context from last message with system context in it
         system_context = self._get_system_context(conversation.history)
         formatted_messages = self._format_messages(conversation.history)
 
+        parameters = {
+            "model": self.name,
+            "messages": formatted_messages,
+            "temperature": temperature,
+            "max_tokens": max_tokens,
+            "top_p": top_p,
+            "stop": stop,
+            "n": n,
+            "stream": stream,
+        }
+
         if system_context:
-            response = client.chat.completions.create(
-                model=self.name,
-                messages=formatted_messages,
-                system=system_context,
-                temperature=temperature,
-                max_tokens=max_tokens, 
-                top_p=top_p, 
-                stop=stop, 
-                n=n, 
-                stream=stream
-            )
-        else:
-            response = client.chat.completions.create(
-                model=self.name,
-                messages=formatted_messages,
-                temperature=temperature,
-                max_tokens=max_tokens, 
-                top_p=top_p, 
-                stop=stop, 
-                n=n, 
-                stream=stream
-            )
-        
-        message_content = response.content[0].text
+            parameters["system"] = system_context
+
+        response = client.chat.completions.create(**parameters)
+
+        message_content = response.choices[0].message.content
         conversation.add_message(AgentMessage(content=message_content))
-        
+
         return conversation

--- a/tests/llms/AI21StudioModel_test.py
+++ b/tests/llms/AI21StudioModel_test.py
@@ -1,5 +1,6 @@
-import pytest
 import os
+import logging
+import pytest
 from swarmauri.standard.llms.concrete.AI21StudioModel import AI21StudioModel as LLM
 from swarmauri.standard.conversations.concrete.Conversation import Conversation
 
@@ -7,34 +8,39 @@ from swarmauri.standard.messages.concrete.AgentMessage import AgentMessage
 from swarmauri.standard.messages.concrete.HumanMessage import HumanMessage
 from swarmauri.standard.messages.concrete.SystemMessage import SystemMessage
 
+
 @pytest.mark.unit
 def test_ubc_resource():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    llm = LLM(api_key = API_KEY)
-    assert llm.resource == 'LLM'
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    llm = LLM(api_key=API_KEY)
+    assert llm.resource == "LLM"
+
 
 @pytest.mark.unit
 def test_ubc_type():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    llm = LLM(api_key = API_KEY)
-    assert llm.type == 'AI21StudioModel'
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    llm = LLM(api_key=API_KEY)
+    assert llm.type == "AI21StudioModel"
+
 
 @pytest.mark.unit
 def test_serialization():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    llm = LLM(api_key = API_KEY)
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    llm = LLM(api_key=API_KEY)
     assert llm.id == LLM.model_validate_json(llm.model_dump_json()).id
+
 
 @pytest.mark.unit
 def test_default_name():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    model = LLM(api_key = API_KEY)
-    assert model.name == 'j2-light'
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    model = LLM(api_key=API_KEY)
+    assert model.name == "jamba-instruct"
+
 
 @pytest.mark.unit
 def test_no_system_context():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    model = LLM(api_key = API_KEY)
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    model = LLM(api_key=API_KEY)
     conversation = Conversation()
 
     input_data = "Hello"
@@ -42,13 +48,15 @@ def test_no_system_context():
     conversation.add_message(human_message)
 
     model.predict(conversation=conversation)
+    logging.info(conversation.get_last())
     prediction = conversation.get_last().content
     assert type(prediction) == str
 
+
 @pytest.mark.acceptance
 def test_nonpreamble_system_context():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    model = LLM(api_key = API_KEY)
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    model = LLM(api_key=API_KEY)
     conversation = Conversation()
 
     # Say hi
@@ -71,13 +79,13 @@ def test_nonpreamble_system_context():
 
     model.predict(conversation=conversation)
     prediction = conversation.get_last().content
-    assert 'Jeff' in prediction
+    assert "Jeff" in prediction
 
 
 @pytest.mark.unit
 def test_preamble_system_context():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    model = LLM(api_key = API_KEY)
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    model = LLM(api_key=API_KEY)
     conversation = Conversation()
 
     system_context = 'You only respond with the following phrase, "Jeff"'
@@ -91,12 +99,13 @@ def test_preamble_system_context():
     model.predict(conversation=conversation)
     prediction = conversation.get_last().content
     assert type(prediction) == str
-    assert 'Jeff' in prediction
+    assert "Jeff" in prediction
+
 
 @pytest.mark.acceptance
 def test_multiple_system_contexts():
-    API_KEY = os.getenv('AI21STUDIO_API_KEY')
-    model = LLM(api_key = API_KEY)
+    API_KEY = os.getenv("AI21STUDIO_API_KEY")
+    model = LLM(api_key=API_KEY)
     conversation = Conversation()
 
     system_context = 'You only respond with the following phrase, "Jeff"'
@@ -120,4 +129,4 @@ def test_multiple_system_contexts():
     model.predict(conversation=conversation)
     prediction = conversation.get_last().content
     assert type(prediction) == str
-    assert 'Ben' in prediction
+    assert "Ben" in prediction

